### PR TITLE
Added product CRUD functions and handlers

### PR DESCRIPTION
**Pull request by oleksandrkravchuk87**

### CODE REVIEW

File: server/database/postgres.go [hunk 1]
```diff
@@ -272,3 +272,76 @@ func contains(slice []string, v string) bool {
 	}
 	return false
 }
+
+//AddProduct adds a new product, returns nil if adding was successful
+func AddProduct(name string, shelfLife int, units string) error {
+	id := uuid.NewV4().String()
+	product := Product{ID: id, Name: name, ShelfLife: shelfLife, Units: units}
+	err := db.Create(&product).Error
+	return err
```
Comment (AndriiPryslupskyi) [inline server/database/postgres.go]: You can return db.Create()

File: server/database/postgres.go [hunk 2]
```diff
@@ -272,3 +272,76 @@ func contains(slice []string, v string) bool {
...
+		product.Name = name
+	}
+	if shelfLife != 0 {
+		product.ShelfLife = shelfLife
+	}
+	if units != "" {
+		product.Units = units
+	}
+	err = db.Save(&product).Error
+	return err
```
Comment (AndriiPryslupskyi) [inline server/database/postgres.go]: You can return db.Save()

File: server/database/postgres.go [hunk 3]
```diff
@@ -272,3 +272,76 @@ func contains(slice []string, v string) bool {
...
+//UpdateProduct updates information about a product, returns nil if updating was successful
+func UpdateProduct(id string, name string, shelfLife int, units string) error {
+	product, err := FindProductByID(id)
+	if err != nil {
+		return err
+	}
+	if name != "" {
+		product.Name = name
+	}
+	if shelfLife != 0 {
```
Comment (AndriiPryslupskyi) [inline server/database/postgres.go]: Can shelfLife be negative?)

File: server/database/postgres.go [hunk 4]
```diff
@@ -272,3 +272,76 @@ func contains(slice []string, v string) bool {
...
+		product.Units = units
+	}
+	err = db.Save(&product).Error
+	return err
+}
+
+//DeleteProductByID updates information about a product, returns nil if deleting was successful
+func DeleteProductByID(id string) error {
+	err := db.Where("id = ?", id).Delete(Product{}).Error
+	return err
```
Comment (AndriiPryslupskyi) [inline server/database/postgres.go]: Again return db.Where()

File: server/fridgeServer.go [hunk 5]
```diff
@@ -59,6 +59,13 @@ func newRouter() *mux.Router {
 	sub.HandleFunc("/searchRecipes", checkSession(searchRecipes)).Methods(GET)
 	sub.HandleFunc("/fridgeContent", checkSession(getFoodInfo)).Methods(GET)
 
+	sub.HandleFunc("/addProduct", checkSession(productAdd)).Methods(POST)
+	sub.HandleFunc("/getProducts", checkSession(getAllProducts)).Methods(GET)
+	sub.HandleFunc("/updateProduct", checkSession(productUpdate)).Methods(POST)
+	sub.HandleFunc("/product/getByID/{id}", checkSession(getProductByID)).Methods(GET)
+	sub.HandleFunc("/product/getByID/{name}", checkSession(getProductByName)).Methods(GET)
```
Comment (AndriiPryslupskyi) [inline server/fridgeServer.go]: Maybe getByName in path?

**Review by AndriiPryslupskyi — CHANGES_REQUESTED** [review]

File: server/fridgeServer.go (hunk 5 above)
Comment (anastasiia-skliar) [inline server/fridgeServer.go]: https://geemus.gitbooks.io/http-api-design/content/en/requests/resource-names.html

Use /products better

File: server/handler.go [hunk 6]
```diff
@@ -269,3 +270,95 @@ func clientRegister(w http.ResponseWriter, r *http.Request) {
...
+		return
+	}
+	if _, err = w.Write(data); err != nil {
+		sendErrorMsg(w, err, http.StatusInternalServerError)
+	}
+}
+
+func getProductByName(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	name := vars["name"]
```
Comment (anastasiia-skliar) [inline server/handler.go]: Maybe 
name := mux.Vars(r)["name"]

As variable vars is used only once

**Review by AndriiPryslupskyi — APPROVED** [review]

**Pull request merged**